ios-base/DroidKaigi 2020/Session/Detail/SessionDetailViewController.swift

```diff
@@ -0,0 +1,204 @@
+    @IBOutlet private weak var intendedAudienceLabel: UILabel!
+
+    @IBOutlet private weak var speakersSection: UIView!
+    @IBOutlet private weak var speakersContainer: UIStackView!
+
+    @IBOutlet private weak var docsSection: UIView!
+    @IBOutlet private weak var videoButton: UIButton!
+    @IBOutlet private weak var slideButton: UIButton!
+
+    @IBOutlet private weak var shareButton: UIBarButtonItem!
+    @IBOutlet private weak var placeButton: UIBarButtonItem!
+    @IBOutlet private weak var eventButton: UIBarButtonItem!
+
+    // MARK: Life cycles
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        bind()
+        loadSession(session)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        navigationController?.navigationBar.barTintColor = .white
+        navigationController?.navigationBar.tintColor = .black
+    }
+}
+
+// MARK: - Private functions
+
+// MARK: Create view
+
+private extension SessionDetailViewController {
+    func loadSession(_ session: Session) {
+        // FIXME: How to detect current lang?
+        titleLabel.text = session.title.ja
+        dateLabel.text = "\(session.startDayText) \(session.startTimeText)-"
+        timeRoomLabel.text = session.timeRoomText
+        descriptionLabel.text = session.desc
+        readMoreButton.isHidden = !descriptionLabel.isTruncated
+
+        switch session {
+        case let speech as SpeechSession:
+            loadSpeechSession(speech)
+        case let service as ServiceSession:
+            loadServiceSession(service)
+        default: break
+        }
+        print(session)
+    }
+
+    func loadSpeechSession(_ session: SpeechSession) {
+        // FIXME: How to detect current lang?
+        categoryLabel.text = session.category.name.ja
+        langLabel.text = session.lang.text.ja
+
+        if let intendedAudience = session.intendedAudience {
+            intendedAudienceLabel.text = intendedAudience
+        } else {
+            audienceSection.isHidden = true
+        }
+        if session.speakers.isEmpty {
+            speakersSection.isHidden = true
+        } else {
+            session.speakers.forEach(addSpeaker(_:))
+        }
+        if let videoUrl = session.videoUrl.flatMap(URL.init) {
+            videoButton.isEnabled = true
+            disposeBag.insert(
+                videoButton.rx.tap
+                    .bind(onNext: { [unowned self] in self.openURL(videoUrl) })
+            )
+        } else {
+            videoButton.isEnabled = false
+        }
+        if let slideUrl = session.slideUrl.flatMap(URL.init) {
+            slideButton.isEnabled = true
+            disposeBag.insert(
+                videoButton.rx.tap
+                    .bind(onNext: { [unowned self] in self.openURL(slideUrl) })
+            )
+        } else {
+            slideButton.isEnabled = false
+        }
+    }
+
+    func loadServiceSession(_ session: ServiceSession) {
```

Comment:
Is this argument required ? 🤔

Reply:
Because `func loadSession(_ session: Session)` would be fat function, splited for readability.
(I'm sorry but I couldn't read the gist, is this correct answer for your question?)

Reply:
Oh, sorry. I wanted to ask about this function argument's need (`session`).
I asked this question because this function does not use this argument.

Reply:
I see. This is OK 😄